semantics/src/lib.rs

```diff
@@ -33,10 +34,15 @@ pub enum Location {
 }
 
 /// Operations that need to be made available during runtime.
-#[derive(Clone, Debug, PartialEq)]
+#[derive(Clone, Debug, PartialEq, PartialOrd, Ord, Eq)]
 pub enum RuntimeOperations {
     /// Encode a set of fixed sized values.
     AbiEncode { params: Vec<FixedSize> },
+    /// Decode a value.
+    AbiDecode {
+        param: FixedSize,
+        loc: AbiDecodeLocation,
```

Comment:
When I first read the usage of `loc` somewhere further down I got confused for a second, thinking about "lines of code". Thoughts on renaming this to `location`?